test/endpoint_tests/test_graphql.py

```diff
@@ -2,54 +2,87 @@
+    ]
 
     db.connect_db()
-    Camps.create(id=1, organisation_id=1, name="some text1", currencyname="hello")
-    Camps.create(id=2, organisation_id=1, name="some text1", currencyname="hello")
-    Camps.create(id=3, organisation_id=1, name="some text1", currencyname="hello")
-    Camps.create(id=4, organisation_id=1, name="some text1", currencyname="hello")
-    Camps.create(id=5, organisation_id=1, name="some text1", currencyname="hello")
+
+    for camp in camps:
+        Camps.create(
+            id=camp["id"],
+            organisation_id=camp["organisation_id"],
+            name=camp["name"],
+            currencyname=camp["currencyname"],
+        )
+
     db.close_db(None)
 
-    graph_ql_query_string = """query { \
-                allBases { \
-                    id \
-                    organisation_id \
-                    name \
-                } \
+    graph_ql_query_string = """query {
+                allBases {
+                    id
+                    organisation_id
+                    name
+                    currencyname
+                }
             }"""
 
     data = {"query": graph_ql_query_string}
     response_data = client.post("/graphql", json=data)
     assert response_data.status_code == 200
-    assert response_data.json["data"]["allBases"][0]["id"] == 1
+    all_bases = response_data.json["data"]["allBases"]
+    for camp in camps:
+        camper = get_base_from_graphql(camp["id"], all_bases)
+        assert camper["id"] == camp["id"]
+        assert camper["organisation_id"] == camp["organisation_id"]
+        assert camper["name"] == camp["name"]
+        assert camper["currencyname"] == camp["currencyname"]
 
 
 def test_base(client):
     """Verify base GraphQL query endpoint"""
+    camps = [
+        {"id": 1, "name": "oak-tree", "organisation_id": 1, "currencyname": "pound"},
+        {"id": 2, "name": "chicken", "organisation_id": 1, "currencyname": "peanuts"},
+        {"id": 3, "name": "sofa", "organisation_id": 1, "currencyname": "candles"},
+    ]
 
     db.connect_db()
-    Camps.create(id=1, organisation_id=1, name="some text1", currencyname="hello")
-    Camps.create(id=2, organisation_id=1, name="some text1", currencyname="hello")
-    Camps.create(id=3, organisation_id=1, name="some text1", currencyname="hello")
-    Camps.create(id=4, organisation_id=1, name="some text1", currencyname="hello")
-    Camps.create(id=5, organisation_id=1, name="some text1", currencyname="hello")
-    db.close_db(None)
 
-    graph_ql_query_string = """query Base {
-                base(id: "1") {
+    for camp in camps:
+        Camps.create(
+            id=camp["id"],
+            organisation_id=camp["organisation_id"],
+            name=camp["name"],
+            currencyname=camp["currencyname"],
+        )
+
+    db.close_db(None)
+    test_id = 1
+    graph_ql_query_string = (
+        """query Base {
+                base(id: "%s") {
```

Comment:
Is there a specific reason you use %-formatting over f-string formatting?
With the latter you can write
```
f"""query Base {
    base(id: "{test_id}") {
        ...
"""
```
which makes it more clear which variable is inserted into the string, instead of having a specifier (%s) first, and then the variable listed at the end.

Reply:
I did look at f strings. I couldn't get it working because of the { brackets within the string. I'll have a look and see if I can make it slightly more explicit.

Reply:
Ah I see what you mean, did not take that into account.